block_carousel.php

```diff
@@ -143,6 +143,11 @@ public function get_content() {
                 continue;
             }
 
+            // Filter any files that are not present or broken.
+            if ($data->heightres === null && $data->widthres === null && $data->contenttype === 'image') {
```

Comment:
There is an `is_null` check you can use, instead of `=== null` which might make it easier to read?

If the property might not be defined at any given point, it might be safer to use `!isset` instead as well.